commit 96d36ad95b03c89857d405b3317efb0188ac59cb
Author: Oliver Upton <oliver.upton@linux.dev>
Date:   Mon Apr 22 20:01:58 2024 +0000

    KVM: selftests: Add stress test for LPI injection
    
    Now that all the infrastructure is in place, add a test to stress KVM's
    LPI injection. Keep a 1:1 mapping of device IDs to signalling threads,
    allowing the user to scale up/down the sender side of an LPI. Make use
    of the new VM stats for the translation cache to estimate the
    translation hit rate.
    
    Since the primary focus of the test is on performance, you'll notice
    that the guest code is not pedantic about the LPIs it receives. Counting
    the number of LPIs would require synchronization between the device and
    vCPU threads to avoid coalescing and would get in the way of performance
    numbers.
    
    Signed-off-by: Oliver Upton <oliver.upton@linux.dev>
    Link: https://lore.kernel.org/r/20240422200158.2606761-20-oliver.upton@linux.dev
    Signed-off-by: Marc Zyngier <maz@kernel.org>

diff --git a/tools/testing/selftests/kvm/Makefile b/tools/testing/selftests/kvm/Makefile
index 4335e5744cc6..e78cac712229 100644
--- a/tools/testing/selftests/kvm/Makefile
+++ b/tools/testing/selftests/kvm/Makefile
@@ -158,6 +158,7 @@ TEST_GEN_PROGS_aarch64 += aarch64/smccc_filter
 TEST_GEN_PROGS_aarch64 += aarch64/vcpu_width_config
 TEST_GEN_PROGS_aarch64 += aarch64/vgic_init
 TEST_GEN_PROGS_aarch64 += aarch64/vgic_irq
+TEST_GEN_PROGS_aarch64 += aarch64/vgic_lpi_stress
 TEST_GEN_PROGS_aarch64 += aarch64/vpmu_counter_access
 TEST_GEN_PROGS_aarch64 += access_tracking_perf_test
 TEST_GEN_PROGS_aarch64 += arch_timer